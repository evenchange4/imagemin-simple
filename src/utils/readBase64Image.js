// @flow

import sharp from 'sharp';
import { type Metadata, type Base64Image } from './type.flow';

const RESIZE_WIDTH = 30;

const readBase64Image = async (imagePath: string): Promise<Base64Image> => {
  const image = sharp(imagePath);
  const { format, width, height }: Metadata = await image.metadata();
  const buffer = await image.resize(RESIZE_WIDTH).toBuffer();

  return {
    imagePath,
    width,
    height,
    format,
    base64: `data:image/${format};base64,${buffer.toString('base64')}`,
  };
};

export default readBase64Image;
