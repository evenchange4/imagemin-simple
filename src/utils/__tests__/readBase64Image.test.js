// @flow
import path from 'path';
import readBase64Image from '../readBase64Image';

it('should return base64Image correctly', async () => {
  const imagePath = path.resolve(
    process.cwd(),
    './test-content1/project1/images/ai-image.jpg',
  );
  const base64Image = await readBase64Image(imagePath);
  expect(base64Image).toMatchSnapshot();
});
