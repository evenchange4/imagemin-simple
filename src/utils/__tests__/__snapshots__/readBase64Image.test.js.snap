// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return base64Image correctly 1`] = `
Object {
  "base64": "data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAANAB4DASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAQYEBQcI/8QAJhAAAgEDAQcFAAAAAAAAAAAAAQIDAAQRBQYSExQxQVEhUpHB8P/EABUBAQEAAAAAAAAAAAAAAAAAAAQF/8QAGxEBAAICAwAAAAAAAAAAAAAAAQACAyESIjH/2gAMAwEAAhEDEQA/AOfo7aQIjvjhsxXI81ZPFHDDGzMDxBld3xQtMSo0AAXcQEP1J7/dSb6AchGSSRGhKgeg64pldSPc5ewaPcRW2qWdxMhliimSRo/cAwJFOW3OoW9/rIutMjaOFkwyFcFWJzj93pCsXIniYdVYH4rQNubjltH0iZYoTLcRJK7lBvElT4pGN0sn5zsAT//Z",
  "format": "jpeg",
  "height": 449,
  "imagePath": "/Users/mtk07866/code/GitHub/imagemin-simple/test-content1/project1/images/ai-image.jpg",
  "width": 1000,
}
`;
